### PR TITLE
Add recipe for sphinx-autoapi

### LINKED ISSUE #6107: Package sphinx-autoapi
Would be good to have a package for [`sphinx-autoapi`]( https://pypi.org/project/sphinx-autoapi/ ), an automatic API doc generation extension for use in conjunction with Sphinx.

### PR DESCRIPTION
**Pull request by CurtLH**
Closes #6107

### CODE REVIEW

**Comment by CurtLH** [comment]
[PR to add license to source tarball](https://github.com/rtfd/sphinx-autoapi/pull/141)

File: recipes/sphinx-autoapi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - wheel ==0.24.0
+    - sphinx
+    - sphinxcontrib-golangdomain
+    - sphinxcontrib-dotnetdomain
+    - unidecode
+
+test:
+  requires:
+    - pydocstyle
+    - salvus_mesher_lite  # [osx]
```
Comment (CurtLH) [inline recipes/sphinx-autoapi/meta.yaml]: Can I use `# [osx]` here, even though I'm trying to build this as `noarch: python`?

**Comment by CurtLH** [comment]
@conda-forge/staged-recipes — this one is ready for review.

**Comment by scopatz** [comment]
Thanks @CurtLH!

File: recipes/sphinx-autoapi/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipes/sphinx-autoapi/meta.yaml]: Given it is in the test section, think we are ok. It's not going to impact the final package (just the build steps). Do you mind raising this point to conda-smithy so we can look at adjust the linter to accept this?

**Pull request merged**